cppapi/client/devapi_base.cpp

```diff
@@ -5512,7 +5492,31 @@ void DeviceProxy::write_pipe(DevicePipe &dev_pipe)
     }
 
     dev_pipe.get_root_blob().reset_insert_ctr();
+}
+
+DevPipeData
+DeviceProxy::createDevPipeData(DevicePipe &dev_pipe) const
+{
+    DevPipeData result;
+
+    result.name = dev_pipe.get_name().c_str();
+    const string &bl_name = dev_pipe.get_root_blob().get_name();
+    if (bl_name.size() != 0)
+        result.data_blob.name = bl_name.c_str();
+
+
+    DevVarPipeDataEltArray *tmp_ptr = dev_pipe.get_root_blob().get_insert_data();
+    if (tmp_ptr == Tango_nullptr)
+    {
+        Except::throw_exception(API_PipeNoDataElement, "No data in pipe!", "DeviceProxy::write_pipe()");
+    }
+
+    DevULong max, len;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define each identifier in a dedicated statement. ([cpp:SingleDeclarationPerStatement](https://sonarcloud.io/organizations/tango-controls/rules#rule_key=cpp%3ASingleDeclarationPerStatement))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.tango-controls%3Acpp-tango%3Atango-9-lts&pullRequest=477&issues=AWWPm0Fu3Qu1svVwsabB&open=AWWPm0Fu3Qu1svVwsabB)

[ISSUE_KEY]: AWWPm0Fu3Qu1svVwsabB